nova/virt/vmwareapi/vmops.py

```diff
@@ -1535,6 +1533,18 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                 self._session._call_method(self._session.vim,
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
+                if vm_group:
+                    cluster_config = self._session._call_method(
+                        vutil, "get_object_property", self._cluster,
+                        "configurationEx")
+                    for group in cluster_config.group:
+                        if group.name == vm_group.name:
+                            if not hasattr(group, 'vm'):
+                                cluster_util.delete_vm_group(
+                                    self._session, cluster.obj,
+                                    group)
+                                break
```

Comment:
Going with the logic above, the `break` should be done one indent less, i.e. when the group matching the name is found, not when the group did not have an attribute `vm`.

Reply:
This comment is still valid. We don't have to iterate further over the groups if we already found the one with the right name - doesn't matter if we took action on it or not.

Reply:
This is still valid ...